File: scripts/new_ashwini/model/reaction_rate.m
% Include electrolyte_potential.m
run('electrolyte_potential.m')

% Local surface reaction rate in negative electrode
function result = j_n(x, d_n_1, d_n_2, I)
    result = I*exp(d_n_1*(x/l_n - 1/d_n_2))/(a_n*F*l_n);
end

% Local surface reaction rate in positive electrode
function result = j_p(x, d_p_1, d_p_2, I)
    result = -I*exp(d_p_1*((L-x)/l_p - 1/d_p_2))/(a_p*F*l_p);
end

% Average surface reaction rate in negative electrode
function result = j_n_bar(I)
    result = I/(a_n*F*l_n);
end

% Average surface reaction rate in positive electrode
function result = j_p_bar(I)
    result = -I/(a_p*F*l_p);
end

File: src/ashwini/model/reaction_rate.jl
include("electrolyte_potential.jl")

"""Local surface reaction rate in negative electrode"""
function jₙ(x, dₙ₁, dₙ₂, I)
    I*exp(dₙ₁*(x/lₙ - 1/dₙ₂))/(aₙ*F*lₙ)
end

"""Local surface reaction rate in positive electrode"""
function jₚ(x, dₚ₁, dₚ₂,I)
    -I*exp(dₚ₁*((L-x)/lₚ - 1/dₚ₂))/(aₚ*F*lₚ)
end

"""Average surface reaction rate in negative electrode """
function j̅ₙ(I)
    I/(aₙ*F*lₙ)
end

"""Average surface reaction rate in positive electrode """
function j̅ₚ(I)
    -I/(aₚ*F*lₚ)
end
